API.md

```diff
@@ -1,12 +1,120 @@
 # Authmosphere API Documentation
 
+## Token Cache
+
+Class to request and cache tokens on client-side.
+
+#### Usage
+
+```typescript
+
+// TokenCacheOptions
+const cacheOptions = {
+  cacheConfig: {
+    percentageLeft: 0.75
+  },
+  logger: someExternalLogger
+};
+
+const oAuthConfig = {
+  grantType: OAuthGrantType.CLIENT_CREDENTIALS_GRANT,
+  accessTokenEndpoint: 'https://....',
+  credentialsDir: './credentialsDir'
+};
+
+const tokenConfig = {
+  'service-foo': ['foo.read', 'foo.write'],
+  'service-bar': ['bar.read']
+};
+
+// create a new cache
+const tokenCache = new TokenCache(tokenConfig, oAuthConfig, cacheOptions);
+
+// Get a token from cache
+tokenCache
+  .get('service-foo') // Needs to match with a key from 'tokenConfig'
+  .then((token: Token) => {
+    console.log(token.access_token);
+  });
+```
+
+### constructor
+
+#### Signature
+
+```ts
+constructor(tokenConfig, oauthConfig: TokenCacheOAuthConfig, options)
+```
+
+#### Arguments
+
+* `tokenConfig: [key: string]: string[]` - the token cache is able to handle multiple tokens. The string array should contain the scopes attached to the token.
```

Comment:
Sometimes we start the section after `-` with uppercase, sometimes with lowercase, lets do it consistent.